src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,73 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppSelector } from '../../app/hook';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { AppDispatch } from '../../app/store';
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const dispatch = useDispatch<AppDispatch>();
+  const currentTodo = useAppSelector(state => state.currentTodoReducer);
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
 
-      <Loader />
+  useEffect(() => {
+    setIsLoading(true);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+    if (currentTodo) {
+      getUser(currentTodo?.userId)
+        .then(setUser)
+        .finally(() => setIsLoading(false));
+    }
+  }, [dispatch, currentTodo]);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  const handleCloseModal = () => {
+    dispatch(currentTodoSlice.actions.removeCurrentTodo(null));
```

Comment:
The `removeCurrentTodo` action is being dispatched with `null` as an argument. If this action is intended to reset the current todo, ensure that the reducer handles `null` appropriately. Alternatively, consider dispatching the action without arguments if the reducer is designed to handle it that way.